pkg/handler/handler.go

```diff
@@ -68,39 +66,43 @@ func NewOpenAPIService(spec *spec.Swagger) (*OpenAPIService, error) {
 	return o, nil
 }
 
-func (o *OpenAPIService) getSwaggerBytes() ([]byte, string, time.Time, error) {
+func (o *OpenAPIService) getSwaggerBytes() ([]byte, string, error) {
 	o.rwMutex.RLock()
 	defer o.rwMutex.RUnlock()
 	specBytes, err := o.jsonCache.Get()
 	if err != nil {
-		return nil, "", time.Time{}, err
+		return nil, "", err
 	}
 	etagBytes, err := o.etagCache.Get()
 	if err != nil {
-		return nil, "", time.Time{}, err
+		return nil, "", err
 	}
-	return specBytes, string(etagBytes), o.lastModified, nil
+	return specBytes, string(etagBytes), nil
 }
 
-func (o *OpenAPIService) getSwaggerPbBytes() ([]byte, string, time.Time, error) {
+func (o *OpenAPIService) getSwaggerPbBytes() ([]byte, string, error) {
 	o.rwMutex.RLock()
 	defer o.rwMutex.RUnlock()
 	specPb, err := o.protoCache.Get()
 	if err != nil {
-		return nil, "", time.Time{}, err
+		return nil, "", err
 	}
 	etagBytes, err := o.etagCache.Get()
 	if err != nil {
-		return nil, "", time.Time{}, err
+		return nil, "", err
 	}
-	return specPb, string(etagBytes), o.lastModified, nil
+	return specPb, string(etagBytes), nil
 }
 
-func (o *OpenAPIService) UpdateSpec(openapiSpec *spec.Swagger) (err error) {
+func (o *OpenAPIService) UpdateSpecLazy(specGetter func() (*spec.Swagger, error)) error {
```

Comment:
Can you add a godoc comment explaining what is going on here, when the function will be evaluated, etc?